src/apps/backend/modules/access-token/rest-api/access-token-controller.ts

```diff
@@ -1,29 +1,18 @@
-import { NextFunction, Request, Response } from 'express';
-
+import { applicationController, Request, Response } from '../../application';
 import AccessTokenService from '../access-token-service';
-import { AccessToken, CreateAccessTokenParams } from '../types';
+import { CreateAccessTokenParams } from '../types';
+
+import { serializeAccessToken } from './access-token-serializer';
 
-export default class AccessTokenController {
-  public static async createAccessToken(
-    req: Request,
-    res: Response,
-    next: NextFunction,
-  ): Promise<void> {
-    try {
-      const { username, password }: CreateAccessTokenParams = req.body as CreateAccessTokenParams;
-      const params: CreateAccessTokenParams = { username, password };
-      const accessToken = await AccessTokenService.createAccessToken(params);
-      res.send(AccessTokenController.serializeAccessTokenAsJSON(accessToken));
-    } catch (e) {
-      next(e);
-    }
-  }
+export const createAccessToken = applicationController(async (
```

Comment:
This seems a bit counter intuitive while reading.

Can you elaborate a bit more on the thinking behind it?

Also the code within `applicationController` is bit involved and not easy to reason with IMO.

Reply:
Yeah agreed - The current definitions of controllers does not _looks_ right.

On what and why is `applicationController` - This mainly takes care of async controllers, so controllers don't have to do `try/catch`. They can simply define their async functions in there and `applicationController` will take care of handling promise and reporting error if any.

Point can be made to make the name more explicit, according to usage, something like `asyncController`. But was thinking what if we add some defacto logic for controllers in there, something like reporting validation errors before running the actual controller.

This will have all the controller's common logic that every controller can use, hence the name - `applicationController`.

Reply:
Would something like `requestHandler` be more appropriate?